File: map/store-multi-values-on-a-map.py
data = (
    ("Toyota", "Avensis"),
    ("Toyota", "Auris"),
    ("Toyota", "Corolla"),
    ("Audi", "A3"),
)

cars = {}
for brand, model in data:
    cars.setdefault(brand, [])
    cars[brand].append(model)

for model in cars.get("Toyota", []):
    print(model)

File: map/store-multi-values-on-a-map.cpp
vector< pair<string, string> > data {
    make_pair("Toyota", "Avensis"),
    make_pair("Toyota", "Auris"),
    make_pair("Toyota", "Corolla"),
    make_pair("Audi", "A3"),
    make_pair("Audi", "A4")
};

multimap<string, string> cars;
for(auto x: data) {
    cars.insert(x);
}

const auto range = cars.equal_range("Toyota");
for(auto x=range.first; x!=range.second; ++x) {
    cout << x->second << endl;
}
